File: cs/BootCampCode/testcases/chapterSix/AdjustPersonalInfoTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support.UI;

namespace BootCampCode.testcases.chapterSix
{
    [TestClass]
    public class AdjustPersonalInfoTest
    {
        [TestMethod]
        public void AdjustPersonalInfo ()
        {
            IWebDriver driver = new ChromeDriver();

            WebDriverWait myWaitVar = new WebDriverWait(driver, TimeSpan.FromSeconds(20));

            // Open the website
            driver.Navigate().GoToUrl("https://techblog.polteq.com/testshop/index.php");

            driver.Manage().Window.Maximize();

            if (driver.FindElement(By.ClassName("login")).Displayed)
            {
                // Click on the login link
                driver.FindElement(By.ClassName("login")).Click();

                // Enter username
                driver.FindElement(By.Id("email")).SendKeys("bootcamper@feelthepain.com");

                // Enter password
                driver.FindElement(By.Id("passwd")).SendKeys("1qazxsw2");

                // Click the Log in button
                driver.FindElement(By.Id("SubmitLogin")).Click();

                //Click the personal info icon
                driver.FindElement(By.CssSelector(".icon-user")).Click();

                //Retrieve the current mail adres
                String currentFirstName = driver.FindElement(By.Id("firstname"))
                    .GetAttribute("value");
                Console.WriteLine(currentFirstName);
                String changedName;
                if (currentFirstName.Equals("Sergeant"))
                {
                    driver.FindElement(By.Id("firstname")).Clear();
                    driver.FindElement(By.Id("firstname")).SendKeys("General");
                    changedName = "General";
                }
                else if (currentFirstName.Equals("General"))
                {
                    driver.FindElement(By.Id("firstname")).Clear();
                    driver.FindElement(By.Id("firstname")).SendKeys("Sergeant");
                    changedName = "Sergeant";
                }
                else
                {
                    driver.FindElement(By.Id("firstname")).Clear();
                    driver.FindElement(By.Id("firstname")).SendKeys("Seargent");
                    changedName = "Seargent";
                }

                driver.FindElement(By.Id("old_passwd")).SendKeys("1qazxsw2");
                driver.FindElement(By.CssSelector("button[name='submitIdentity']")).Click();

                //Check if the empty message appears
                Assert.AreEqual("Your personal information has been successfully updated."
                    , driver.FindElement(By.CssSelector(".alert.alert-success")).Text
                    , "Validate if succesfull message appears");

                driver.FindElement(By.CssSelector(".account>span")).Click();

                //Click the personal info icon
                driver.FindElement(By.CssSelector(".icon-user")).Click();

                //Check the new name
                Console.WriteLine(driver.FindElement(By.Id("firstname")).GetAttribute("value"));
                Assert.AreEqual(changedName, driver.FindElement(By.Id("firstname")).GetAttribute("value")
                    , "Validate name change");
            }
        }
    }
}

File: java/src/test/java/testcases/chapterSix/AdjustPersonalInfoTest.java
package testcases.chapterSix;

import org.assertj.core.api.Assertions;
import org.openqa.selenium.By;
import org.testng.annotations.Test;
import testcases.TestShopScenario;


public class AdjustPersonalInfoTest extends TestShopScenario {

    @Test
    public void adjustPersonalInfo() throws InterruptedException {

        driver.manage().window().maximize();

        if(driver.findElement(By.className("login")).isDisplayed()){
            // Click on the login link
            driver.findElement(By.className("login")).click();

            // Enter username
            driver.findElement(By.id("email")).sendKeys("bootcamper@feelthepain.com");

            // Enter password
            driver.findElement(By.id("passwd")).sendKeys("1qazxsw2");

            // Click the Log in button
            driver.findElement(By.id("SubmitLogin")).click();

            //Click the personal info icon
            driver.findElement(By.cssSelector(".icon-user")).click();

            //Retrieve the current mail adres
            String currentFirstName = driver.findElement(By.id("firstname")).getAttribute("value");
            System.out.println(currentFirstName);
            String changedName;
            if (currentFirstName.equals("Sergeant")){
                driver.findElement(By.id("firstname")).clear();
                driver.findElement(By.id("firstname")).sendKeys("General");
                changedName = "General";
            }
            else if (currentFirstName.equals("General")){
                driver.findElement(By.id("firstname")).clear();
                driver.findElement(By.id("firstname")).sendKeys("Sergeant");
                changedName = "Sergeant";
            }
            else{
                driver.findElement(By.id("firstname")).clear();
                driver.findElement(By.id("firstname")).sendKeys("Seargent");
                changedName = "Seargent";
            }

            driver.findElement(By.id("old_passwd")).sendKeys("1qazxsw2");
            driver.findElement(By.cssSelector("button[name='submitIdentity']")).click();

            //Check if the empty message appears
            Assertions.assertThat(driver.findElement(By.cssSelector(".alert.alert-success")).getText())
                    .as("Validate if succesfull message appears")
                    .isEqualTo("Your personal information has been successfully updated.");

            driver.findElement(By.cssSelector(".account>span")).click();

            //Click the personal info icon
            driver.findElement(By.cssSelector(".icon-user")).click();

            //Check the new name
            //By running the test twice the console outputs should switch!
            System.out.println(driver.findElement(By.id("firstname")).getAttribute("value"));
            Assertions.assertThat(driver.findElement(By.id("firstname")).getAttribute("value"))
                    .as("Validate name change")
                    .isEqualTo(changedName);
        }
        else
        {
            //Assertion or other failsafe. "If" can be removed but this way the bootcampers can try to build a fail
            //safe. Reasoning here is: test shouldn't run if the login is not shown
        }
    }
}
